drivers/hmis/app/graphql/mutations/update_form_definition.rb

```diff
@@ -35,5 +36,41 @@ def resolve(id:, input:)
         { errors: errors }
       end
     end
+
+    private
+
+    def convert_form_definition(definition)
+      {
+        item: definition['item'].map do |i|
+          convert_form_item(i)
+        end,
+      }
+    end
+
+    def convert_form_item(item)
+      converted = basic_convert(item)
+      converted['mapping'] = basic_convert(item['mapping']) if item['mapping']
+
+      if item['item']
+        converted['item'] = item['item'].map do |i|
+          convert_form_item(i)
+        end
+      end
+
+      converted
+    end
+
+    def basic_convert(input)
+      converted = {}
+
+      input.keys.each do |key|
+        next if key == '__typename'
+        next if input[key].nil?
+
+        converted[key.underscore] = input[key]
+      end
+
+      converted
+    end
```

Comment:
This method could be simplified a bit by using rails hash methods:
```
      input.excluding('__typename').
        compact. # drop keys with nil values
        transform_keys(&:underscore) # snake_case keys
```

I'd also suggest renaming `basic_convert` to indicate that it is for hashes specifically, maybe `transform_hash` or something.

It would also be possible to use `.deep_transform_keys(&:underscore)` which can be called once and transform the keys on the whole definition. But we need to traverse it anyway to do the `compact`, so doing the key transformation here seems fine. There is no handy helper for doing a ["deep compact"](https://stackoverflow.com/questions/56457020/ruby-how-do-i-remove-nil-empty-values-from-a-nested-hash-and-why-arent-my-atte). Doing it the way you have -- with separate cases for transforming `mapping` and any other nested objects -- seems like a fine approach to me!

Reply:
Wow beautiful! Thanks for the simplification suggestion!